src/components/layouts/BRCharts.tsx

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import { Line } from 'react-chartjs-2';
+
+export interface BRChartOpts {
+  xAxisValues: any[];
+  yAxisValues: any[];
+  fill: boolean;
+  lineTension: number;
+  backgroundColor: string;
+  borderColor: string;
+  pBorderColor: string;
+  pBackgroundColor: string;
+  pBorderWidth: number;
+  pHoverBorderWidth: number;
+  pHoverRadius: number;
+  pRadius: number;
+  label: string;
+  pHoverBackgroundColor: string;
+  pHoverBorderColor: string;
+}
+
+// override xAxisValues, yAxisValues, label, backgroundColor, borderColor, pHoverBackgroundColor, pHoverBorderColor
+export let chartDefaultOptValues: BRChartOpts = {
+  xAxisValues: [],
+  yAxisValues: [],
+  fill: false,
+  lineTension: 0.1,
+  backgroundColor: 'rgba(75,192,192,0.4)',
+  borderColor: 'rgba(75,192,192,1)',
+  pBackgroundColor: '#fff',
+  pBorderColor: 'rgba(75,192,2,1)',
+  pBorderWidth: 1,
+  pHoverBorderWidth: 2,
+  pHoverRadius: 5,
+  pRadius: 1,
+  label: '',
+  pHoverBackgroundColor: 'rgba(75,192,192,1)',
+  pHoverBorderColor: 'rgba(220,220,220,1)'
+};
+
+interface BRChartProps {
+  opts: BRChartOpts[];
+}
+
+interface Datasets {
+  label: string;
+  fill: boolean;
+  lineTension: number;
+  backgroundColor: string;
+  borderColor: string;
+  pointBorderColor: string;
+  pointBackgroundColor: string;
+  pointBorderWidth: number;
+  pointHoverRadius: number;
+  pointHoverBackgroundColor: string;
+  pointHoverBorderColor: string;
+  pointHoverBorderWidth: number;
+  pointRadius: number;
+  data: any[];
+}
+
+interface BRChartOptionsFormated {
+  labels: any[];
+  datasets: Datasets[];
+}
+
+interface BRChartState {
+  wrapperClassName: string;
+  chartJSOpts: any;
+}
+
+export class BRCharts extends React.Component<BRChartProps, BRChartState> {
+  constructor(props: BRChartProps) {
+    super(props);
+    console.log('props is');
+    console.log(this.props.opts);
+    this.state = {
+      chartJSOpts: null,
+      wrapperClassName: 'canvas-chart-wrapper'
+    };
+  }
+
```

Comment:
@Harkishen-Singh Its better to use a prop "type" for this component, and then we can use conditional statements to render charts. (type here can be "line". "bar", etc)

Reply:
@ganeshpatro321 sure. Will do as said.

Reply:
@ganeshpatro321 actually, we plan to use as a line chart only. time-series values are best suited for line-charts (also area charts, can be converted by `fill: true` property). Hence. the `className` is just used to provide a wrapper style to the component, and hence a state.

Defining as a prop would require the parent component to sent the className which will always be same for every screen, and hence a redundancy. What are your views?

Reply:
Well if we are sure that we will only use line charts, then its good to go. @Harkishen-Singh